tests: Bluetooth: df: Unit tests failing due to missing configuration and wrong preparation of connection object

ppryga-nordic opened the pull request:
Wrong feature was enabled for a connection object as one the currently
used features. The enabled feature was CTE response. In case if the
feature exchange has not happened, it is verified if CTE request feature
is enabled, not the CTE response feature. The problem is in a common
code that prepares connection object for unit tests.

The commit fixes enabled features preparation for a connection object.
Implementation of ll_df_set_cl_cte_tx_params was updated to verify
if the number of requested CTE in a periodic advertising chain is less
or equal to maximum supported value that is provided by CONFIG_BT_CTLR-
_DF_PER_ADV_CTE_NUM_MAX Kconfig option. In the past the value was
verified against maximum value allowed by BT Core Spec Vol 4, Part E,
section 7.8.80. Check against BT Core spec provided value could lead to
memory overwrite.

The unit tests start to fail because default value of the CONFIG_BT_-
CTLR_DF_PER_ADV_CTE_NUM_MAX is 1, hence is shall be changed in unit test
configuration file.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
tests: Bluetooth: df: Wrong feature set for features_used

Wrong feature was enabled for a connection object as one the currently
used features. The enabled feature was CTE response. In case if the
feature exchange has not happened, it is verified if CTE request feature
is enabled, not the CTE response feature. The problem is in a common
code that prepares connection object for unit tests.

The commit fixes enabled features preparation for a connection object.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga-nordic committed:
tests: Bluetooth: df: Missing conf for max num of CTE in per adv chain

Implementation of ll_df_set_cl_cte_tx_params was updated to verify
if the number of requested CTE in a periodic advertising chain is less
or equal to maximum supported value that is provided by CONFIG_BT_CTLR-
_DF_PER_ADV_CTE_NUM_MAX Kconfig option. In the past the value was
verified against maximum value allowed by BT Core Spec Vol 4, Part E,
section 7.8.80. Check against BT Core spec provided value could lead to
memory overwrite.

The unit tests start to fail because default value of the CONFIG_BT_-
CTLR_DF_PER_ADV_CTE_NUM_MAX is 1, hence is shall be changed in unit test
configuration file.

The commit adds missing configuration.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

Thalley commented on tests/bluetooth/df/common/src/bt_conn_common.c:
```diff
@@ -44,7 +44,7 @@ uint16_t ut_bt_create_connection(void)
 #endif /* CONFIG_BT_CTLR_DF_CONN_CTE_REQ */
 
-	conn->llcp.fex.features_used |= BIT(BT_LE_FEAT_BIT_CONN_CTE_RESP);
+	conn->llcp.fex.features_used |= BIT(BT_LE_FEAT_BIT_CONN_CTE_REQ);
```
Based on the name `BT_LE_FEAT_BIT_CONN_CTE_REQ`, should this be moved inside the `CONFIG_BT_CTLR_DF_CONN_CTE_REQ` guard?

ppryga replied:
I've moved it in a guard, to be sure it is set only when the CTE request is compiled in.

ppryga-nordic committed:
tests: Bluetooth: df: remove unnecessary testcase.yaml

The testcase.yaml file in tests/bluetooth/df directory makes
twister trying to build tests that are not available.
All tests in the df subdirectory have private testcase.yaml
files.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

ppryga commented:
I've added small fixes and new commit that removes `testcase.yaml` from `tests/bluetooth/df` directory.
